src/data/scripts/internet/main.py

```diff
@@ -148,29 +151,61 @@ def process_chunk_data(chunk):
 
 
 def filter_field(data, source):
-    data["source"] = source
-    meta_dict = {"filename": args.input_file.split("/")[-1]}
     del data["log_pp_score"], data["prediction"]
+    if source != "oscar_cl":
+        data["source"] = source
+        meta_dict = {"filename": args.input_file.split("/")[-1]}
+        if source == "mc4":
+            data["created_date"] = str(data["timestamp"])
+            meta_dict["url"] = data["url"]
+            del data["timestamp"], data["url"]
+
+        elif source == "cc100":
+            data["created_date"] = "2020-10-23T23:31:11.000Z"
+
+        elif source == "oscar":
+            data["created_date"] = "2023-06-08T14:30:28.000Z"
+            data["source_id"] = data["id"]
+            del data["id"]
+        data["meta"] = str(meta_dict)
 
-    if source == "mc4":
-        data["created_date"] = str(data["timestamp"])
-        meta_dict["url"] = data["url"]
-        del data["timestamp"], data["url"]
-
-    elif source == "cc100":
-        data["created_date"] = "2020-10-23T23:31:11.000Z"
-
-    elif source == "oscar":
-        data["created_date"] = "2023-06-08T14:30:28.000Z"
-        data["source_id"] = data["id"]
-        del data["id"]
-
-    data["meta"] = str(meta_dict)
     if data["updated_date"] == "None":
         data["updated_date"] = data["created_date"]
     return data
 
 
+def read_jsonl_zst_files(dir_path):
+    for root, _, files in os.walk(dir_path):
+        for filename in files:
+            if filename.endswith(".jsonl.zst"):
+                file_path = os.path.join(root, filename)
+                try:
+                    with zstd.open(open(file_path, "rb"), "rt", encoding="utf-8") as f:
+                        id = 0  # Initialize ID for each file
+                        for row in f:
+                            item = json.loads(row)
+                            yield {
+                                "text": item["content"],
```

Comment:
Is this missing some keys?
<img width="753" alt="image" src="https://github.com/OpenThaiGPT/openthaigpt-pretraining/assets/12471844/0c25e758-6f5a-4710-99ea-cc0c3cdbbb33">

Reply:
updated_date will create in process_chunk_data(chunk) line 87